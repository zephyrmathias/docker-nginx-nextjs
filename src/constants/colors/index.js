// primary
export const black = '#000'
export const black6 = '#0d0e0e'
export const black5 = '#252729'
export const black4 = '#3d4144'
export const black3 = '#555a5f'
export const black2 = '#6e7479'

export const gray8 = '#747476'
export const gray7 = '#8e8e8f'
export const gray6 = '#a7a7a9'
export const gray5 = '#c1c1c2'
export const gray4 = '#dbdbdb'
export const gray3 = '#f4f4f5'
export const white = '#fff'

export const blue8 = '#164593'
export const blue7 = '#2362B7'
export const blue6 = '#3384DB'
export const blue5 = '#47A9FF'
export const blue4 = '#75C5FF'
export const blue3 = '#90D7FF'
export const blue2 = '#B5E8FF'

// secondary

export const green8 = '#147223'
export const green7 = '#1F8E28'
export const green6 = '#2EAA2E'
export const green5 = '#4AC63F'
export const green4 = '#7EDC6B'
export const green3 = '#A4ED8C'
export const green2 = '#CBF9B5'

export const yellow8 = '#937600'
export const yellow7 = '#B79500'
export const yellow6 = '#DBB600'
export const yellow5 = '#FFD800'
export const yellow4 = '#FFE43F'
export const yellow3 = '#FFEC66'
export const yellow2 = '#FFF499'

export const red8 = '#93112F'
export const red7 = '#B71C31'
export const red6 = '#DB2832'
export const red5 = '#FF3E38'
export const red4 = '#FF7A69'
export const red3 = '#FF9F87'
export const red2 = '#FFC6AF'
