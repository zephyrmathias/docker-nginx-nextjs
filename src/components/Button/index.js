import styled from 'styled-components'
import { yellow7, yellow5, gray3 } from '../../constants/colors'

function getButtonType(type) {
  switch (type) {
    case 'secondary':
      return 'background: black; color: white;'
    default:
      return `
        background: ${yellow5};
        color: black;

        &:hover {
          background: ${yellow7};
        }

        &:disabled {
          background: ${gray3};
        }
      `
  }
}

const Button = styled.button`
  border-radius: 4px;
  height: 44px;
  font-size: 14px;
  line-height: 14px;
  outline: none;
  user-select: none;
  border: none;
  text-transform: uppercase;
  min-width: 150px;
  cursor: pointer;
  ${({ type }) => getButtonType(type)}
`

Button.defaultProps = {
  type: 'primary',
}

export default Button
