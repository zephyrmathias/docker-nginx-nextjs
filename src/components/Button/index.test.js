import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import Button from './index'
import { yellow7, yellow5, gray3 } from '../../constants/colors'

test('Button should apply default values', () => {
  const component = shallow(<Button />)
  expect(component).toMatchSnapshot()
  expect(component).toHaveStyleRule('border-radius', '4px')
  expect(component).toHaveStyleRule('height', '44px')
  expect(component).toHaveStyleRule('font-size', '14px')
  expect(component).toHaveStyleRule('line-height', '14px')
  expect(component).toHaveStyleRule('outline', 'none')
  expect(component).toHaveStyleRule('user-select', 'none')
  expect(component).toHaveStyleRule('border', 'none')
  expect(component).toHaveStyleRule('text-transform', 'uppercase')
  expect(component).toHaveStyleRule('min-width', '150px')
  expect(component).toHaveStyleRule('cursor', 'pointer')
  expect(component).toHaveStyleRule('background', yellow5)
  expect(component).toHaveStyleRule('background', yellow7, {
    modifier: ':hover ',
  })
  expect(component).toHaveStyleRule('background', gray3, {
    modifier: ':disabled ',
  })
  expect(component).toHaveStyleRule('color', 'black')
})

test('Button should apply secondary styles', () => {
  const component = shallow(<Button type="secondary" />)
  expect(component).toMatchSnapshot()
  expect(component).toHaveStyleRule('border-radius', '4px')
  expect(component).toHaveStyleRule('height', '44px')
  expect(component).toHaveStyleRule('font-size', '14px')
  expect(component).toHaveStyleRule('line-height', '14px')
  expect(component).toHaveStyleRule('outline', 'none')
  expect(component).toHaveStyleRule('user-select', 'none')
  expect(component).toHaveStyleRule('border', 'none')
  expect(component).toHaveStyleRule('text-transform', 'uppercase')
  expect(component).toHaveStyleRule('min-width', '150px')
  expect(component).toHaveStyleRule('cursor', 'pointer')
  expect(component).toHaveStyleRule('background', 'black')
  expect(component).toHaveStyleRule('color', 'white')
})
