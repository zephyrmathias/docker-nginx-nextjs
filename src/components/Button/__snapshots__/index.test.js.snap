// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should apply default values 1`] = `
.c0 {
  border-radius: 4px;
  height: 44px;
  font-size: 14px;
  line-height: 14px;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  text-transform: uppercase;
  min-width: 150px;
  cursor: pointer;
  background: #FFD800;
  color: black;
}

.c0:hover {
  background: #B79500;
}

.c0:disabled {
  background: #f4f4f5;
}

<button
  className="c0"
  type="primary"
/>
`;

exports[`Button should apply secondary styles 1`] = `
.c0 {
  border-radius: 4px;
  height: 44px;
  font-size: 14px;
  line-height: 14px;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  text-transform: uppercase;
  min-width: 150px;
  cursor: pointer;
  background: black;
  color: white;
}

<button
  className="c0"
  type="secondary"
/>
`;
